mavimage/GPS.py

```diff
@@ -0,0 +1,49 @@
+from pymavlink import mavutil
+import sys
+from optparse import OptionParser
+from dronekit import connect, VehicleMode
+
+class GPS:
+
+    def __init__(self, connection):
+        parser = OptionParser("readdata.py [options]")
+        parser.add_option("--baudrate", dest="baudrate", type='int',
+                          help="master port baud rate", default=115200)
+        parser.add_option("--device", dest="device", default=None, help="serial device")
+        parser.add_option("--rate", dest="rate", default=4, type='int', help="requested stream rate")
+        parser.add_option("--source-system", dest='SOURCE_SYSTEM', type='int', default=255,
+                          help='MAVLink source system')
+        parser.add_option("--showmessages", dest="showmessages", action='store_true',
+                          help="show incoming messages", default=False)
+        (opts, args) = parser.parse_args()
+
+        if opts.device is None:
+            print("Please select a serial device")
+            sys.exit(1)
+        self.connection = mavutil.mavlink_connection(opts.device, baud=opts.baudrate)
```

Comment:
I am not 100% sure but I think storing a connection will cause a circular dependency and thus the garbage collector will not be able to free the memory (memory leak).